clap_man/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+#![doc(html_logo_url = "https://raw.githubusercontent.com/clap-rs/clap/master/assets/clap.png")]
+#![doc = include_str!("../README.md")]
+#![warn(missing_docs, trivial_casts, unused_allocation, trivial_numeric_casts)]
+#![forbid(unsafe_code)]
+
+mod man;
+mod render;
+
+use std::io::Write;
+
+/// Manpage sections, the most common is [`Section::Executable`].
+#[derive(Debug, Clone, Copy)]
+pub enum Section {
+    /// Executable programs or shell commands
+    Executable,
+    /// System calls (functions provided by the kernel)
+    SystemCalls,
+    /// Library calls (functions within program libraries)
+    LibraryCalls,
+    /// Special files (usually found in /dev)
+    SpecialFiles,
+    /// File formats and conventions, e.g. /etc/passwd
+    FileFormats,
+    /// Games
+    Games,
+    /// Miscellaneous (including macro packages and conventions), e.g. man(7), groff(7)
+    Miscellaneous,
+    /// System administration commands (usually only for root)
+    SystemAdministrationCommands,
+    /// Kernel routines [Non standard]
+    KernelRoutines,
+}
+
+impl Section {
+    fn value(&self) -> i8 {
+        match self {
+            Section::Executable => 1,
+            Section::SystemCalls => 2,
+            Section::LibraryCalls => 3,
+            Section::SpecialFiles => 4,
+            Section::FileFormats => 5,
+            Section::Games => 6,
+            Section::Miscellaneous => 7,
+            Section::SystemAdministrationCommands => 8,
+            Section::KernelRoutines => 9,
+        }
+    }
+}
+
+/// Man page generator
+#[derive(Debug, Clone)]
+pub struct Man {
+    section: Option<Section>,
+    manual: Option<String>,
+    sections: Vec<(String, Vec<String>)>,
+}
+
+impl Default for Man {
+    fn default() -> Self {
+        Self {
+            section: Some(Section::Executable),
+            manual: Some("General Commands Manual".to_string()),
+            sections: Vec::new(),
+        }
+    }
+}
+
+/// Generate manpage for your application using the most common default values.
+pub fn generate_manpage<'a>(app: &mut clap::App<'a>, buf: &mut dyn Write) {
+    let man = Man::default();
+    man.render(app, buf);
+}
+
+impl Man {
+    /// Create a new builder for man pages.
+    pub fn new() -> Self {
+        Man {
+            section: None,
+            manual: None,
+            sections: Vec::new(),
+        }
+    }
```

Comment:
Why do `new` and `default` differ?

Reply:
Maybe we should rename `new` to `builder` or something like that? My thoughts were that `default` should be what most users ought to need, while `new` is used when you want more control.

Reply:
imp `::new()` and `::default()` should be the same.  If its good for one, its good for the other.  

When first getting into Rust, I wondered about the semantics of `default` and over time I've come to feel that `default` should not mean "fill this out with a default policy" but "give me a no-arg constructor".

Reply:
I think my opinion is closer to what the Rust documentation says about the `Default` trait:

> A trait for giving a type a useful default value.

Though I'm not married to the current implementation.

Reply:
How that is applied is still left open.

I would like to see `new` and `default` be consistent.